dace/libraries/blas/nodes/dot.py

```diff
@@ -155,6 +160,137 @@ def expansion(node, state, sdfg):
         return tasklet
 
 
+
+
+@dace.library.expansion
+class ExpandDOTFPGAStreamingLinearReduction(ExpandTransformation):
+
+    environments = []
+
+    @staticmethod
+    def make_sdfg(
+            dtype,
+            partial_width,
```

Comment:
What exactly is the partial width, and how does the user choose this?

Reply:
It defines the width of the buffer for the inner reduction. In pseudo python code:
```
partial = [0 for _ in range(partial_width)]
for i in range(n//veclen):
    x_buf = x.pop() # shape = [veclen]
    y_buf = y.pop() # shape = [veclen]
    for j in range(veclen):
        partial[j % partial_width] += x_buf[j] + y_buf[j]

res = 0
for i in range(partial_width):
    res += partial[i]

return res
```

So, the inner most map reduces into a buffer of size `partial_width`. This is to reduce the width of the final reduction.

It is set by the user, as a parameter to the library node.

Reply:
So the size of this is hardware-dependent, right? It has to be larger than the latency of the accumulation operation. What I meant with "how does the user choose this": it seems hard for the user to know what to set this to.

Is 2 really enough for Xilinx? I thought you needed more like 8. I would rather set this to something conservative so we know it always works, and then let the user reduce it if they want

Reply:
It should be a performance/resource trade-off. For optimal performance, I'd assume it should be equal to `veclen` and then the final reduction should be a reduction tree.